_code/InSituTree/R/supervisedSubcluster.R

```diff
@@ -80,9 +80,9 @@ supervisedSubcluster <- function(reference_profiles,
   }
   
   # Run supervised InSituType on cell and gene subet
-  sup_res <- insitutypeML(x[nonZeroCount_idx, use_genes],
```

Comment:
Would this trigger warning from `insitutypeML()` every time you ran it since many genes in counts are not presented in reference profiles? An alternative is to calculate `bg` in your `runInSituTree()` wrapper function when `neg` is not `NULL` but `bg` is and then pass the `bg` to subsequent child function.

Reply:
You can do this with `InSituType:::estimateBackground()`[ function](https://github.com/Nanostring-Biostats/InSituType/blob/main/R/utilities.R#L9) which would use `bg` if it's not `NULL`.

Reply:
I've updated the function to calculate `bg` on the whole dataset prior to any subclustering. I no longer get a warning about genes in the counts matrix missing from the profile.